src/Azure/Shared/Storage/AzureTableDataManager.cs

```diff
@@ -48,23 +48,25 @@ internal class AzureTableDataManager<T> where T : class, ITableEntity, new()
         /// <summary> Connection string for the Azure storage account used to host this table. </summary>
         protected string ConnectionString { get; set; }
 
-        private CloudTable tableReference;
+        public CloudTable Table { get; private set; }
 
-        public CloudTable Table => tableReference;
+        public AzureStoragePolicyOptions StoragePolicyOptions;
 
         /// <summary>
         /// Constructor
         /// </summary>
         /// <param name="tableName">Name of the table to be connected to.</param>
         /// <param name="storageConnectionString">Connection string for the Azure storage account used to host this table.</param>
         /// <param name="logger">Logger to use.</param>
-        public AzureTableDataManager(string tableName, string storageConnectionString, ILogger logger)
+        /// <param name="storagePolicyOptions">Optional Storage Policy Configuration.</param>
+        public AzureTableDataManager(string tableName, string storageConnectionString, ILogger logger, AzureStoragePolicyOptions storagePolicyOptions = default)
```

Comment:
This really should not allow a `null` value for the policy object. It should throw if it's null.

The same goes for every time a default parameter value is allowed.